lucene/expressions/src/java/org/apache/lucene/expressions/js/JavascriptCompiler.java

```diff
@@ -191,9 +220,21 @@ private Expression compileExpression(ClassLoader parent) throws ParseException {
 
     try {
       generateClass(getAntlrParseTree(), classWriter, externalsMap);
-
-      final Class<? extends Expression> evaluatorClass = new Loader(parent)
-        .define(COMPILED_EXPRESSION_CLASS, classWriter.toByteArray());
+      
+      final Class<? extends Expression> evaluatorClass;
```

Comment:
Just a suggestion. My personal preference is to set up a supplier (or a function) statically, depending on conditions, then just use that later forever. In this case you could have something like: 
````
static Function<byte[], Class<? extends Expression>> classLoader;
static {
  if (MH_...) {
    classLoader = (bytes) -> {};
  } else {
    classLoader = (bytes) -> {};
  }
}
````
which would be defined once and used without the knowledge of its internal implementation details later on, forever. Just a thought.

Reply:
I was thinking about that, but as it throws a checked Exception it would need a new functional interface. I moved the whole stuff to a separate method `defineClass()`.